src/main/java/org/wildfly/installationmanager/spi/InstallationManager.java

```diff
@@ -119,4 +120,22 @@ public interface InstallationManager {
      * @throws Exception
      */
     Path createSnapshot(Path targetPath) throws Exception;
+
+    /**
+     * Generate an apply update CLI command.
+     * The generated command can be run in separate process to apply changes.
+     *
+     * @param candidatePath - path to the update candidate
+     * @return empty Optional if the CLI is not available, or a CLI command.
```

Comment:
Could you describe a bit this case? when the CLI won't be available to return an empty Optional?

Reply:
Basically if prospero-cli is not available, e.g if a project only uses prospero-common.

Arguably that's a bit of a corner case, I'm not sure if it'll ever come into play.

Reply:
ok, it sounds like it will never happen on the server integration. 

An exception could be more appropriate that an Optional since the use of this method is to generate the launch command, but I left it to your criteria.

Reply:
Makes sense, I changed to exceptions, let me know if that looks better